common.blocks/demo/demo.js

```diff
@@ -54,9 +69,25 @@ modules.define('demo', [ 'i-bem__dom', 'pretty', 'functions__debounce' ], functi
                 return;
             }
 
-            this._html.setValue(pretty(bemhtml.apply(BEMJSON), {
-                max_char: 1000
-            }));
+            var compiledText = '';
+
+            switch(this.params.engine) {
+                case 'vidom':{
+                    compiledText = JSON.stringify(bemhtml.apply(BEMJSON), null, 4);
+                    break;
+                }
+                case 'bemhtml':{
+                    compiledText = pretty(bemhtml.apply(BEMJSON), { max_char: 1000});
+                    break;
+                }
+                case 'bemjson':{
+                    compiledText = JSON.stringify(BEMJSON, null, 4);
```

Comment:
Yeah, much better. Even better could be a way with classes of Engine interface for each engine. Like:
```js
class BEMXJSTEngine {
  render(bemjson) {
    throw new Error('Doesn\'t .');
  }
  _safe(fn) {
    try {
      return fn();
    } catch(e) {
      return engineName.toUpperCase() + ' error: ' + e.message + '\n' + e.stack;
    }
  }
}
// ...
class BEMHTMLEngine extends BEMXJSTEngine {
  render(bemjson) {
    this.compile(bemjson);
    this.exportApply(bemhtml);
    return this._safe(function() {
      return pretty(bemhtml.apply(BEMJSON), {max_char: 1000});
    });
  }
}
```